pkg/mutate/webhook_test.go

```diff
@@ -0,0 +1,617 @@
+				Name:      "offloading",
+				Namespace: "test",
+			},
+			Spec: offv1alpha1.NamespaceOffloadingSpec{
+				NamespaceMappingStrategy: offv1alpha1.EnforceSameNameMappingStrategyType,
+				PodOffloadingStrategy:    offv1alpha1.RemotePodOffloadingStrategyType,
+				ClusterSelector: corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"A,B"},
+						}, {
+							Key:      "provider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"C,D"},
+						}, {
+							Key:      "NotProvider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+				},
+				},
+			},
+		}, {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "offloading",
+				Namespace: "test",
+			},
+			Spec: offv1alpha1.NamespaceOffloadingSpec{
+				NamespaceMappingStrategy: offv1alpha1.EnforceSameNameMappingStrategyType,
+				PodOffloadingStrategy:    offv1alpha1.LocalAndRemotePodOffloadingStrategyType,
+				ClusterSelector: corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"A,B"},
+						}, {
+							Key:      "provider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "region",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"C,D"},
+						}, {
+							Key:      "NotProvider",
+							Operator: corev1.NodeSelectorOpExists,
+						},
+						},
+					},
+				},
+				},
+			},
+		}}
+
+		nodeSelectors := []*corev1.NodeSelector{
+			{NodeSelectorTerms: []corev1.NodeSelectorTerm{}},
+			{NodeSelectorTerms: []corev1.NodeSelectorTerm{
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "region",
+						Operator: corev1.NodeSelectorOpIn,
+						Values:   []string{"A,B"},
+					}, {
+						Key:      "provider",
+						Operator: corev1.NodeSelectorOpExists,
+					}, {
+						Key:      liqoconst.TypeLabel,
+						Operator: corev1.NodeSelectorOpIn,
+						Values:   []string{liqoconst.TypeNode},
+					},
+					},
+				},
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "region",
+						Operator: corev1.NodeSelectorOpIn,
+						Values:   []string{"C,D"},
+					}, {
+						Key:      "NotProvider",
+						Operator: corev1.NodeSelectorOpExists,
+					}, {
+						Key:      liqoconst.TypeLabel,
+						Operator: corev1.NodeSelectorOpIn,
+						Values:   []string{liqoconst.TypeNode},
+					},
+					},
+				},
+			}},
```

Comment:
This could also be moved to a separate function (leveraged by the one above). Similar for the other occurrences of duplicated code below.